jobsAdmin/JobsAdmin.go

```diff
@@ -295,19 +294,45 @@ func (ja *jobsAdmin) JobMgrEnsureExists(jobID common.JobID,
 	return ja.jobIDToJobMgr.EnsureExists(jobID,
 		func() ste.IJobMgr {
 			// Return existing or new IJobMgr to caller
-			return ste.NewJobMgr(ja.concurrency, jobID, ja.appCtx, ja.cpuMonitor, level, commandString, ja.logDir, ja.concurrencyTuner, ja.pacer, ja.slicePool, ja.cacheLimiter, ja.fileCountLimiter, ja.JobLogger)
+			return ste.NewJobMgr(ja.concurrency, jobID, ja.appCtx, ja.cpuMonitor, level, commandString, ja.logDir, ja.concurrencyTuner, ja.pacer, ja.slicePool, ja.cacheLimiter, ja.fileCountLimiter)
 		})
 }
 
-// Same as JobMgrEnsureExists but with addition parameter of user logger.
-func (ja *jobsAdmin) JobMgrCreateWithLogger(jobID common.JobID,
-	level common.LogLevel, commandString string, logger common.ILoggerResetable) ste.IJobMgr {
+// JobMgrCleanup cleans up the jobMgr identified by the given jobId. It undoes what NewJobMgr() does, basically it does the following:
+// 1. Stop all go routines started to process this job.
+// 2. Release the memory allocated for this JobMgr instance.
+// Note: this is not thread safe and only one goroutine should call this for a job.
+func (ja *jobsAdmin) JobMgrCleanUp(jobId common.JobID) bool {
+	// First thing get the jobMgr.
+	jm, found := ja.JobMgr(jobId)
 
-	return ja.jobIDToJobMgr.EnsureExists(jobID,
-		func() ste.IJobMgr {
-			// Return existing or new IJobMgr to caller
-			return ste.NewJobMgr(ja.concurrency, jobID, ja.appCtx, ja.cpuMonitor, level, commandString, ja.logDir, ja.concurrencyTuner, ja.pacer, ja.slicePool, ja.cacheLimiter, ja.fileCountLimiter, logger)
-		})
+	if found {
+		fmt.Println("JobMgrDone Enter")
+		// Delete the jobMgr from jobIDtoJobMgr map, so that next call will fail.
+		ja.DeleteJob(jobId)
+		fmt.Println("Job Deleted in kv")
```

Comment:
do they use the term kv in azcopy?
else, We can say "Job deleted from jobMgr map"